src/index.ts

```diff
@@ -215,10 +215,31 @@ if (ENABLE_GRAPHQL_UPLOAD) {
 }
 
 app.use("/v2", (req, res, next) => {
-  if (req.url !== "/") return next()
+  // This matches all routes that start with '/v2'.
+  //
+  // We only want to support:
+  //   - POSTs -> requests to '/v2'
+  //   - GETs:
+  //     - requests to '/v2' (loads GraphiQL in non-prod mode)
+  //     - requests to '/v2?query=...' (runs a query)
+  if (req.method === "POST") {
+    if (req.url !== "/") return next()
+  } else if (req.method === "GET") {
+    if (req.url !== "/") {
+      if (!req.url.startsWith("/?")) return next()
```

Comment:
If the GET is not strictly to "/v2", then it _must_ be of the form "/v2?query=..." (that is, still to the root and requiring a query param called `query` - this is for GraphiQL support, or for linking to the backing query for a given Force page, something which helps debug those).